pkg/operator/controller/controller_dns_daemonset_test.go

```diff
@@ -497,6 +498,36 @@ func TestDaemonsetConfigChanged(t *testing.T) {
 				}
 				daemonset.Spec.Template.Annotations["cluster-autoscaler.kubernetes.io/enable-ds-eviction"] = "true"
 			},
+			expect: false, // it is already there, and true
+		},
+		{
+			description: "if the enable-ds-eviction annotation is changed",
+			mutate: func(daemonset *appsv1.DaemonSet) {
+				if daemonset.Spec.Template.Annotations == nil {
+					daemonset.Spec.Template.Annotations = map[string]string{}
+				}
+				daemonset.Spec.Template.Annotations["cluster-autoscaler.kubernetes.io/enable-ds-eviction"] = ""
+			},
+			expect: true,
+		},
+		{
+			description: "if the target workload annotation is added",
+			mutate: func(daemonset *appsv1.DaemonSet) {
+				if daemonset.Spec.Template.Annotations == nil {
+					daemonset.Spec.Template.Annotations = map[string]string{}
+				}
+				daemonset.Spec.Template.Annotations["target.workload.openshift.io/management"] = "{\"effect\": \"PreferredDuringScheduling\"}"
+			},
+			expect: false, // it is already there with the same value
```

Comment:
Again, isn't this the same as "if nothing changes"?

Reply:
Fixed